src/picai_prep/dcm2mha.py

```diff
@@ -26,372 +29,389 @@
+        self.patient_id = patient_id
+        self.study_id = study_id
+        self.series: List[Series] = []
+        self.log = [f'Importing {plural(len(paths), "serie")}']
 
-    def _convert(self):
-        convert_count = 0
-        for item in self.valid_items:
-            for _, status in item['targets'].items():
-                convert_count += 1 if status is None else 0
-
-        self.info(f"Converting {convert_count} DICOM series to MHA.")
-
-        # convert all items with valid targets to mha
-        success_count, skip_count = 0, 0
-        targets = []
-        for item in self.valid_items:
-            for target, status in item['targets'].items():
-                if status is None:  # therefore selected
-                    destination = os.path.join(self.output_dir, item['patient_id'], target + '.mha')
-                    if os.path.exists(destination):
-                        item['targets'][target] = 'Skipped, target output path already exists'
-                        self.item_log(item, 'skipped')
-                        skip_count += 1
-                    else:
-                        targets.append((item, target, destination))
+        fullpaths = set()
+        for path in paths:
+            full_path = input_dir / path
+            serie = Series(full_path, patient_id, study_id)
+            try:
+                if not full_path.exists():
+                    raise ArchiveItemPathNotFoundError(path)
+                elif not full_path.is_dir():
+                    raise NotADirectoryError(path)
+                elif path in fullpaths:
+                    raise FileExistsError(path)
+            except Exception as e:
+                serie.error = e
+                logging.error(str(e))
+            else:
+                fullpaths.add(full_path)
+            finally:
+                self.log.append(f'\t+ ({len(self.series)}) {full_path}')
+                self.series.append(serie)
+        if not all([serie.is_valid for serie in self.series]):
+            self.invalidate()
+
+    @property
+    def valid_series(self):
+        return [item for item in self.series if item.is_valid]
+
+    def invalidate(self):
+        for serie in self.valid_series:
+            serie.error = SeriesException('Invalidated due to critical error in sibling')
+
+    def write_log(self, msg: str):
+        self.log.append(msg)
+
+    def compile_log(self):
+        divider = '=' * 120
+        log = [divider,
+               f'CASE {self.patient_id}_{self.study_id}',
+               f'\tPATIENT ID\t{self.patient_id}',
+               f'\tSTUDY ID\t{self.study_id}\n']
+        log += self.log
+        log += ['\nSERIES']
+        for i, serie in enumerate(self.series):
+            log.append(f'({i}) {serie.path.as_posix()}')
+            log.extend([f'\t{l}' for l in serie.log])
+            log.append(f'\tFATAL: {serie.error}\n' if serie.error else '')
+        return '\n'.join(log)
+
+    def convert(self, *args):
```

Comment:
No need for this. Just use `def convert(self, output_dir):`

Reply:
`def convert(self, *output_dir)` would only work. Multiprocessing only accepts `*args` as far as I understand.

Reply:
Partly true. For the `ThreadPoolExecutor` from `concurrent.futures` this works perfectly fine (just tested it to double check.)

Reply:
I prefer the `*args` because it shows (at least for me) that the function is likely the mp entry point, and more easily allows us to add any further variables we may want to add in the future.

I neatly unpack the *args in the line below, so there is good clarity as to what `*args` is expected to be.